elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,941 @@ function xml_form_elements_creative_commons($element, &$form_state) {
     'sg' => t('Singapore'),
     'si' => t('Slovenia'),
     'za' => t('South Africa'),
-    'kr' => t('South Korea'),
     'es' => t('Spain'),
     'se' => t('Sweden'),
     'ch' => t('Switzerland'),
     'tw' => t('Taiwan'),
     'th' => t('Thailand'),
-    'uk' => t('UK: England & Wales'),
-    'scotland' => t('Uk: Scotland'),
     'ug' => t('Uganda'),
-    'us' => t('United states'),
+    'uk' => t('UK: England & Wales'),
+    'scotland' => t('UK: Scotland'),
+    'us' => t('United States'),
+    've' => t('Venezuela'),
     'vn' => t('Vietnam'),
   );
+  return array_merge($international_option, $jurisdiction_options);
+}
 
-  $default_jurisdiction = 'international';
-
-  $commercial_options = array(
-    'y' => t('Yes'),
-    'n' => t('No'),
-  );
-
+/**
+ * Creates a `creative_commons` form element.
+ *
+ * @param array $element
+ *   The form element.
+ * @param array $form_state
+ *   The form state.
+ *
+ * @return array
+ *   The element definition.
+ */
+function xml_form_elements_creative_commons($element, &$form_state) {
   // Get an ID for ajax.
-  if (isset($form_state['storage']['xml_form_elements'][$element['#name']]['license_output_id'])) {
-    $license_output_id = $form_state['storage']['xml_form_elements'][$element['#name']]['license_output_id'];
+  if (isset($form_state['storage']['xml_form_elements'][$element['#name']]['license_fieldset_id'])) {
+    $license_fieldset_id = $form_state['storage']['xml_form_elements'][$element['#name']]['license_fieldset_id'];
   }
   else {
-    $license_output_id = drupal_html_id('license_output');
-    $form_state['storage']['xml_form_elements'][$element['#name']]['license_output_id'] = $license_output_id;
+    $license_fieldset_id = drupal_html_id('license_fieldset');
+    $form_state['storage']['xml_form_elements'][$element['#name']]['license_fieldset_id'] = $license_fieldset_id;
   }
 
-  // Disable Creative commons by default.
-  $disabled = TRUE;
-  // Figure out license state.
-  if (!isset($form_state['input'][$element['#name']]) && isset($element['#default_value'])) {
-    // If there is a value defined then it wasn't disabled.
-    if (!empty($element['#default_value'])) {
-      $disabled = FALSE;
-    }
-
-    // Reversing this array facilitates value_callback mangling by community.
-    $default_value_array = array_reverse(explode('/',
-      str_replace($creative_commons_base_url, "", $element['#default_value'])
-    ));
-    $properties = explode('-', array_pop($default_value_array));
+  // Get current values, process them and store them in the element.
+  $input = isset($form_state['input'][$element['#name']]) ? $form_state['input'][$element['#name']] : NULL;
+  $values = xml_form_elements_creative_commons_process_input($element, $input, $form_state);
+  $element['#value'] = $values;
```

Comment:
Does this mean, that whatever the user does, $element's value will be $value?
I guess you all tested this pretty well, but i was in the impression that input was not to be trusted and to allow multi step to work correctly in a custom element like this you needed to make sure that form_set_value() is called if the value submitted by the users needs to be changed/something to make sure it  persists.  https://api.drupal.org/api/drupal/includes%21form.inc/function/form_set_value/7.x